applications/ConstitutiveLawsApplication/custom_constitutive/structural_elements_constitutive_laws/thickness_integrated_isotropic_constitutive_law.cpp

```diff
@@ -0,0 +1,761 @@
+    ConstitutiveLaw::Parameters& rValues)
+{
+    CalculateMaterialResponseCauchy(rValues);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+
+void ThicknessIntegratedIsotropicConstitutiveLaw::CalculateMaterialResponseCauchy(
+    ConstitutiveLaw::Parameters& rValues)
+{
+    KRATOS_TRY
+
+    // Get Values to compute the constitutive law:
+    Flags& r_flags = rValues.GetOptions();
+    const auto& r_material_properties = rValues.GetMaterialProperties();
+    const IndexType number_of_laws = mConstitutiveLaws.size();
+    const auto subprop_strain_size = mConstitutiveLaws[0]->GetStrainSize(); // 3
+    const auto subprop_dimension = mConstitutiveLaws[0]->WorkingSpaceDimension(); // 2
+
+    // Previous flags saved
+    const bool flag_compute_constitutive_tensor = r_flags.Is(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR);
+    const bool flag_compute_stress = r_flags.Is(ConstitutiveLaw::COMPUTE_STRESS);
+    const bool flag_computed_strain = true;
+
+    std::vector<double> coordinates;
+    std::vector<double> weights;
+
+    CalculateCoordinatesAndWeights(coordinates, weights, mThicknessIntegrationPoints, r_material_properties);
+
+    // The generalized strain vector, constant
+    const Vector generalized_strain_vector = rValues.GetStrainVector(); // size 8
+    Vector generalized_stress_vector(VoigtSize); // size 8
+    Matrix generalized_constitutive_matrix(VoigtSize, VoigtSize); // 8x8
+    generalized_constitutive_matrix.clear();
+    generalized_stress_vector.clear();
+
+    const auto it_prop_begin = r_material_properties.GetSubProperties().begin();
+    Properties &r_subprop = *(it_prop_begin);
+
+    if (flag_compute_stress || flag_compute_constitutive_tensor) {
+
+        // Auxiliary stress vector
+        Vector& r_stress_vector = rValues.GetStressVector(); // size 3
+        Vector& r_strain_vector = rValues.GetStrainVector(); // size 3
+        Matrix& r_constitutive_matrix = rValues.GetConstitutiveMatrix(); // size 3x3
+        r_strain_vector.resize(subprop_strain_size, false);
+        r_stress_vector.resize(subprop_strain_size, false);
+        r_constitutive_matrix.resize(subprop_strain_size, subprop_strain_size, false);
+        r_strain_vector.clear();
+        r_stress_vector.clear();
+        r_constitutive_matrix.clear();
+
+        Matrix F(subprop_dimension, subprop_dimension); // 2x2
+        double weight, z_coord, z_coord2, aux_weight, aux_weight2, detF;
+
+        const double h_max = GetMaxReferenceEdgeLength(rValues.GetElementGeometry());
+        const double alpha = 0.1;
+        const double thickness = r_material_properties[THICKNESS];
+        const double t_square = thickness * thickness;
+        const double stenberg_stabilization = (5.0 / 6.0) * t_square / (t_square + alpha * h_max * h_max);
```

Comment:
this factor 5/6 is for the computation of the effective shear in homogeneous materials. Are u sure it also applies in the heteregenous case?

Reply:
good question. Let's investigate this...